File: Graph/connectedComponents/connectedComponents.js
(function () {
    
    function DFS(edges, n, sv, visited, arr) {
        visited[sv] = true;
        arr.push(sv);
        for (let i=0; i<n; i++) {
            if (i === sv) continue;

            if (edges[sv][i] && !visited[i]) {
                DFS(edges,n,i,visited,arr);
            }
        }
    }   

    function connectedComponents(edges, n) {
        let visited = [];
        for (let i = 0; i<n; i++) visited[i] = false;
        let ans = [];
        for (let i=0; i<n; i++) {
            let arr = [];
            if (!visited[i]) {
                DFS(edges,n,i,visited,arr);
                arr.sort(function(a,b) {
                    if (a < b) {
                        return -1;
                    } else {
                        return 1;
                    }
                });
                ans.push(arr);
            }
        }

        return ans;
    }

    (function main(){

        let n = 9;
        let e = 10;
        
        let connection = [
            {sv:0, ev:8},
            {sv:1, ev:6},
            {sv:1, ev:7},
            {sv:1, ev:8},
            {sv:5, ev:8},
            {sv:6, ev:0},
            {sv:7, ev:3},
            {sv:7, ev:4},
            {sv:8, ev:7},
            {sv:2, ev:5},
        ];

        let edges = [];
        for (let i = 0; i<n; i++) {
            edges[i] = [];
            for (let j = 0; j<n; j++) {
                edges[i][j] = false;
            }
        }

        for (let i = 0; i<e; i++) {
            let {sv,ev} = connection[i];
            edges[sv][ev] = true;
            edges[ev][sv] = true;
        }

        let ans = connectedComponents(edges,n);
        for(let i of ans) {
            console.log(i);
        }
    }());
}());

File: Graph/connectedComponents/connectedComponents.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using std::cin;
using std::cout;
using std::endl;
using std::vector;
using std::sort;

void DFS(bool** edges, int n, bool* visited, int sv,vector<int>& arr) {
    visited[sv] = true;
    arr.push_back(sv);
    for (int i{}; i < n; i++) {
        if (i == sv) continue;
        if (edges[sv][i] && !visited[i]) DFS(edges, n, visited, i, arr);
    }
}

vector<vector<int> >* connectedComponents(bool** edges, int n) {
    bool* visited{new bool[n]};
    for(int i{}; i < n; i++) visited[i] = false;
    vector<vector<int> >* ans {new vector<vector<int> >()};
    vector<int> tmp;
    for (int i{}; i < n; i++) {
        if(!visited[i]) {
            DFS(edges,n,visited,i,tmp);
            sort(tmp.begin(), tmp.end());
            ans->push_back(tmp);
            tmp.clear();
        }
    }
    delete [] visited;
    return ans;
}

int main() {
    int n, e;
    cin >> n >> e;

    bool** edges {new bool*[n]};
    for (int i{}; i < n; i++) {
        edges[i] = new bool[n];
        for (int j{}; j < n; j++) edges[i][j] = false;
    }

    for(int i{}; i < e; i++) {
        int sv, ev;
        cin >> sv >> ev;
        edges[sv][ev] = true;
        edges[ev][sv] = true;
    }

    vector<vector<int> >* ans{connectedComponents(edges, n)};
    for(int i{}; i < ans->size(); i++) {
        vector<int> arr{ans->at(i)};
        for(auto i : arr) cout << i << " ";
        cout << std::endl;
    }

    for(int i{}; i < n; i++) delete [] edges[i];
    delete [] edges;
    delete ans;
}
